src/actions/ratesActions.js

```diff
@@ -35,54 +38,72 @@ import { reportErrorLog } from 'utils/common';
 
 // selectors
 import { accountAssetsPerChainSelector } from 'selectors/assets';
-import { assetsBalancesSelector, supportedAssetsPerChainSelector } from 'selectors';
+import {
+  assetsBalancesSelector,
+  ratesPerChainSelector,
+  supportedAssetsPerChainSelector,
+} from 'selectors';
 
 // models, types
-import type { Rates } from 'models/Asset';
 import type { Dispatch, GetState } from 'reducers/rootReducer';
 import type { Chain } from 'models/Chain';
+import type { RatesBySymbol } from 'models/Rates';
 
 // actions
 import { saveDbAction } from './dbActions';
 
 
-export const setRatesAction = (newRates: Rates) => {
-  return (dispatch: Dispatch, getState: GetState) => {
-    if (isEmpty(newRates)) return;
-    const { rates: { data: currentRates = {} } } = getState();
-    // $FlowFixMe: flow update to 0.122
-    const rates = { ...currentRates, ...newRates };
-    dispatch(saveDbAction('rates', { rates }, true));
-    dispatch({ type: UPDATE_RATES, payload: rates });
+export const setIsFetchingRatesAction = (isFetching: boolean) => ({
+  type: SET_FETCHING_RATES,
+  payload: isFetching,
+});
+
+export const setRatesAction = (
+  chain: Chain,
+  rates: RatesBySymbol,
+) => {
+  return async (dispatch: Dispatch, getState: GetState) => {
+    if (isEmpty(rates)) return;
+
+    dispatch({ type: SET_CHAIN_RATES, payload: { chain, rates } });
+
+    const updatedRatesPerChain = ratesPerChainSelector(getState());
```

Comment:
Pls use raw redux selection here, as using selector could cause serialization problems if selector is changed in the future, which will work for components but not for saveDb

```suggestion
    const updatedRatesPerChain = getState().rates....
```